### PR TITLE
Try to delete graph edges when hotspot assertions change

### PR DESCRIPTION
**Pull request by Vagabond**
Summary
----
- Fix a bunch of stale document fields
- Update documents properly
- Remove HotspotType enum
- Temporary fix new_location oracles bug

### CODE REVIEW

File: src/handler/arangodb.rs [hunk 1]
```diff
@@ -269,9 +264,54 @@ impl DB {
         };
 
         match res {
-            Ok(_) => {
+            Ok(thing) => {
                 tracing::debug!("successfully populated {:?} hotspot", hotspot_type);
```
Comment (JayKickliter) [inline src/handler/arangodb.rs:268]: we need to move this down now

Comment (vihu) [reply src/handler/arangodb.rs:268]: This is fixed in the latest update.

File: src/handler/arangodb.rs [hunk 2]
```diff
@@ -211,73 +206,100 @@ impl DB {
...
+            .bind_var("pub_key", hotspot._key.to_string());
+
+        let aql = aql_builder.build();
+        let res = self.inner.aql_query::<Value>(aql).await;
 
         match res {
-            Ok(_) => {
-                tracing::debug!("successfully populated {:?} hotspot", hotspot_type);
-                Ok(())
+            Ok(upsert_result) => {
```
Comment (JayKickliter) [inline src/handler/arangodb.rs:228]: This march arm has gotten so long it's hard to follow. Can we move the `Err(_)` arms above?

File: src/handler/arangodb.rs [hunk 3]
```diff
@@ -211,73 +206,100 @@ impl DB {
...
+                            Ok(())
+                        }
+                    }
+                    _ => Ok(()),
+                }
             }
             Err(ClientError::Arango(ae)) if [1210, 1200].contains(&ae.error_num()) => {
                 tracing::debug!(
-                    "warning, collection: {:?}, hotspot_type: {:?}, {:?}: {:?}",
+                    "warning, collection: {:?}, {:?}: {:?}",
```
Comment (JayKickliter) [inline src/handler/arangodb.rs:301]: "warning" feels redundant when we could use `warn!` instead

File: src/handler/arangodb.rs (hunk 2 above)
Comment (vihu) [reply src/handler/arangodb.rs:238]: [cec6e1f](https://github.com/novalabsxyz/arango-etl/pull/23/commits/cec6e1fba5ca692282ccc76f79b188f7ca606630)

File: src/handler/arangodb.rs [hunk 4]
```diff
@@ -211,79 +206,106 @@ impl DB {
...
+                        let new_gain = json!(hotspot.gain);
+                        let new_location = json!(hotspot.location);
+                        let new_pubkey = json!(hotspot._key);
+
+                        if old_elevation != Some(&new_elevation)
+                            || old_gain != Some(&new_gain)
+                            || old_location != Some(&new_location)
+                        {
+                            tracing::info!(
+                                "something changed, removing related edges.\nold_pubkey: {:?}\nnew_pubkey:{:?}\nold_elevation: {:?}\tnew_elevation: {:?}\nold_gain: {:?}\tnew_gain: {:?}\nold_location: {:?}\tnew_location: {:?}",
```
Comment (JayKickliter) [inline src/handler/arangodb.rs:261]: pretty please let's keep newlines out of logging

Comment (vihu) [reply src/handler/arangodb.rs:261]: Sigh, it was getting harder to read the values being compared. Any other suggestion to format it?

**Review by JayKickliter — CHANGES_REQUESTED** [review]
The only thing I care about are the newlines in the logging. Other than that merge away when you're confident that it's complete.

File: src/handler/arangodb.rs (hunk 4 above)
Comment (JayKickliter) [reply src/handler/arangodb.rs:261]: Not really other than using grep/sed when viewing the logs

**Review by JayKickliter — APPROVED** [review]

File: src/handler/arangodb.rs (hunk 4 above)
Comment (vihu) [reply src/handler/arangodb.rs]: I've updated it to just log everything separately [6acafc5](https://github.com/novalabsxyz/arango-etl/pull/23/commits/6acafc5f35c2102af890a7a6ace69c3eddb702af)
